lib/IRGen/GenDecl.cpp

```diff
@@ -1462,6 +1462,9 @@ getIRLinkage(const UniversalLinkageInfo &info, SILLinkage linkage,
     return RESULT(External, Hidden, Default);
 
   case SILLinkage::Private: {
+    if (info.forcePublicDecls())
+      return getIRLinkage(info, SILLinkage::SharedExternal, isDefinition,
+                          isWeakImported);
```

Comment:
I'm also not sure whether this is the best possible set of flags. It appears to work...

Reply:
It probably ought to be based on whether this is a definition, so that definitions are shared (or maybe even private) and non-definitions are public-external.